airbyte-integrations/connectors/source-greenhouse/source_greenhouse/greenhouse.yaml

```diff
@@ -46,6 +46,7 @@ definitions:
         inject_into: "path"
       url_base: "*ref(definitions.requester.url_base)"
   base_stream:
+    type: DeclarativeStream
```

Comment:
This is a case worth noting. In this case, we have a custom substream slicer component which references a specific parent stream. For normal components, we append `type: DeclarativeStream` when it is missing, however, we cannot do type inference for custom components because the fields can be dynamic. So in the greenhouse (and other custom substream components), we need to explicitly define the type. We are also shifting in the direction of explicit type definitions for all manifests. Although added verbosity, this additional inference layer does not play as nicely with connector builder UI because it's an added layer of transformation on both sides.

Reply:
That's a good question. Something that I don't love is that existing connectors are inconsistent about usage of `type` vs inferred types. But going back and adding it to every connector will muddle up an already complicated change more. My first thought is we should modify existing templates and docs i this PR to always have type specified. 

And we should go back and add `type` in a follow up PR for connectors written, pre-beta release.

And just to note as to why this affected only `source-greenhouse` in this case is because this is the only case of a custom stream slicer which therefore is unable to infer the `DeclarativeStream` type. All other cases of substream slicer use the first class component which will automatically populate the `type: DeclarativeStream` during the transformation before we do all this.